packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -225,8 +228,21 @@ const DropdownMenuRootContent = React.forwardRef((props, forwardedRef) => {
       portalled={portalled}
       onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
         event.preventDefault();
-        context.triggerRef.current?.focus();
+        if (!preventAutoFocusRef.current) context.triggerRef.current?.focus();
+      })}
+      onEscapeKeyDown={composeEventHandlers(props.onEscapeKeyDown, () => {
+        preventAutoFocusRef.current = false;
       })}
+      onPointerDownOutside={composeEventHandlers(
+        props.onPointerDownOutside,
+        (event) => {
+          const originalEvent = event.detail.originalEvent;
+          const ctrlLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === true;
+          const isRightClick = originalEvent.button === 2 || ctrlLeftClick;
+          preventAutoFocusRef.current = isRightClick || !context.modal;
```

Comment:
I don't like that we need to pass `modal` down in context to make this work. I've had to do this because we have no access to the trigger at the `Menu` level. An alternative is to move the trigger into `Menu` but I'm a little concerned at the sheer number of concerns that `Menu` is already handling.

If anyone can think of an alternative I'd be open to it.

Reply:
It doesn't bother me too much, I think ultimately this is easier to understand than if we had to move around the whole trigger requirement into `Menu`. We've tried that before and it was pretty confusing if I remember correctly.

Reply:
Yeah true, I really didn't really want to go there a third time haha